csharp/ql/src/Diagnostics/DiagnosticNoExtractionErrors.ql

```diff
@@ -0,0 +1,17 @@
+/**
+ * @name Successfully extracted files
+ * @description A list of all files in the source code directory that were extracted
+ *              without encountering an extraction or compiler error in the file.
+ * @kind diagnostic
+ * @id cs/diagnostics/successfully-extracted-files
+ */
+
+import csharp
+import semmle.code.csharp.commons.Diagnostics
+
+from File file
+where
+  file.fromSource() and
+  not exists(ExtractorError e | e.getLocation().getFile() = file) and
+  not exists(CompilerError e | e.getLocation().getFile() = file)
+select file, ""
```

Comment:
Is the `""` needed?

Reply:
Yes, it is needed. This is probably how we indicate that these results are success ones. See also [here](https://github.com/github/codeql/pull/5721/files#diff-7574be8592b66acd793052052ded0ded7b1621ca3f9fff0e76e39800e8763e7fR12-R14).